src/plugins/discover/public/application/components/discover.tsx

```diff
@@ -119,150 +176,161 @@ export function Discover({
           showSearchBar={true}
           useDefaultBehaviors={true}
         />
+        {flyout}
         <main className="dscApp__frame">
-          <EuiResizableContainer>
-            {(EuiResizablePanel, EuiResizableButton) => (
-              <>
-                <EuiResizablePanel
-                  minSize="20%"
-                  initialSize={100}
-                  scrollable={false}
-                  className="dscApp__sidebar"
-                >
-                  <DiscoverSidebar
-                    columns={state.columns}
-                    fieldCounts={fieldCounts}
-                    hits={rows}
-                    indexPatternList={indexPatternList}
-                    onAddField={addColumn}
-                    onAddFilter={onAddFilter}
-                    onRemoveField={onRemoveColumn}
-                    selectedIndexPattern={searchSource && searchSource.getField('index')}
-                    setIndexPattern={setIndexPattern}
-                  />
-                </EuiResizablePanel>
+          <>
+            <div className="dscSidebar dscSidebar__desktop eui-hideFor--xs eui-hidenFor--s">
+              <DiscoverSidebar
+                columns={state.columns}
+                fieldCounts={fieldCounts}
+                hits={rows}
+                indexPatternList={indexPatternList}
+                onAddField={addColumn}
+                onAddFilter={onAddFilter}
+                onRemoveField={onRemoveColumn}
+                selectedIndexPattern={searchSource && searchSource.getField('index')}
+                setIndexPattern={setIndexPattern}
+              />
+            </div>
+            <div className="dscSidebar dscSidebar__mobile eui-showFor--xs eui-showFor--s">
+              <DiscoverSidebarMobile
+                columns={state.columns}
+                fieldCounts={fieldCounts}
+                hits={rows}
+                indexPatternList={indexPatternList}
+                onAddField={addColumn}
+                onAddFilter={onAddFilter}
+                onRemoveField={onRemoveColumn}
+                selectedIndexPattern={searchSource && searchSource.getField('index')}
+                setIndexPattern={setIndexPattern}
+              />
```

Comment:
Wouldn't that be a good case for `EuiShowFor` /  `EuiHideFor`?

https://elastic.github.io/eui/#/utilities/responsive

The current approach has one disadvantage, the rendering is done twice, and one Sidebar is just hidden. With EuiShowFor` /  `EuiHideFor` there was no double rendering. While it won't make much difference to a short field list, there are huge field list out there in the field.

Reply:
Nice catch! I added `EuiShowFor` and `EuiHideFor`.